### ISSUE TITLE
java.lang.IllegalStateException: Cannot schedule view update because there's no initiator to do this

### ISSUE THREAD

**Issue by Azodox** [DevNatan/inventory-framework#151]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### 🐛 Describe the bug

Hey,

I'm not sure if it's an actual bug or do I just miss something? Anyways, I'm trying to open to player a view that updates each seconds, here is my code :

How I open it to my player
```java
      var clan = this.economy.getClanManager().getClan(player);
      if(clan == null){
          player.sendMessage(Component.text("Vous n'avez pas de clan.").color(NamedTextColor.RED));
          return;
      }

      var frame = ViewFrame.of(economy, new ClanView(clan)).register();
      frame.open(ClanView.class, player);
```
How I declare my view
```java
package fr.olten.xmas.clan.view;

import fr.olten.xmas.clan.Clan;
import fr.olten.xmas.ladder.RankingLadder;
import fr.olten.xmas.util.ItemBuilder;
import me.saiintbrisson.minecraft.View;
import me.saiintbrisson.minecraft.ViewContext;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.jetbrains.annotations.NotNull;

import java.time.Duration;

public class ClanView extends View {

    private final Clan clan;

    public ClanView(Clan clan) {
        super(5 * 9, "§6§lInformations §8❱ §r" + clan.getName());

        this.clan = clan;
        this.scheduleUpdate(Duration.ofSeconds(1));

        this.setCancelOnClick(true);
        this.setCancelOnClone(true);
        this.setCancelOnDrag(true);
        this.setCancelOnDrop(true);
        this.setCancelOnMoveOut(true);
        this.setCancelOnPickup(true);
        this.setCancelOnShiftClick(true);

        this.slot(3, 3).onUpdate(update -> update.setItem(new ItemBuilder(Material.BOOK, 1)
                .displayName("§b§lMembres")
                .lore(clan.getPlayers().stream().map(player -> "§7" + Bukkit.getOfflinePlayer(player).getName()).toList())
                .build()));
        this.slot(3, 5).onUpdate(update -> update.setItem(new ItemBuilder(Material.PAPER, 1)
                .displayName("§6§lInformations")
                .lore(
                        "§7Nom : " + clan.getName(),
                        "§7DisplayName : " + MiniMessage.miniMessage().serialize(clan.displayName()),
                        "§7Couleur : " + clan.color(),
                        "§7Propriétaire : " + Bukkit.getOfflinePlayer(clan.getOwner()).getName()
                )
                .build()));
        this.slot(3, 7).onUpdate(update -> update.setItem(new ItemBuilder(Material.LADDER, 1)
                .displayName("§a§lClassement : #" + RankingLadder.getRanking(clan))
                .build()));
        this.slot(5, 4, new ItemBuilder(Material.REDSTONE, 1).displayName("§c§lStatus des options").build());
        this.slot(5, 5, new ItemBuilder(Material.LEGACY_REDSTONE_COMPARATOR, 1).displayName("§c§lGérer les options de clan").build());
    }

    @Override
    protected void onRender(@NotNull ViewContext context) {
        if(this.clan.getOwner().equals(context.getPlayer().getUniqueId())){
            context.slot(2, 5).withItem(
                    new ItemBuilder(Material.TOTEM_OF_UNDYING, 1)
                            .displayName("§e§lVous êtes le propriétaire du clan")
                            .build()
            );
        }
    }

    @Override
    protected void onUpdate(@NotNull ViewContext context) {
        context.getPlayer().sendMessage(Component.text("update"));
    }
}

```

### ✔️ Expected behavior

Just open you please.

### 👣 Steps to Reproduce

1. Create a basic view
2. Call method `scheduleUpdate` in constructor
3. Try to open it

### 💻 Platform

- [X] Bukkit
- [ ] Sponge
- [X] Other

### ⭐ Version

1.19

### ✍️ Additional context

(Still running Paper)

**Comment by Azodox** [comment]
Oh I forgot the error, sorry, here you go :

```
[10:42:12 ERROR]: [Economy] [ACF] Exception in command: clan
[10:42:12 ERROR]: [Economy] [ACF] java.lang.IllegalStateException: Cannot schedule view update because there's no initiator to do this
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//me.saiintbrisson.minecraft.BukkitViewComponentFactory.scheduleUpdate(BukkitViewComponentFactory.java:160)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//me.saiintbrisson.minecraft.AbstractVirtualView.scheduleUpdate(AbstractVirtualView.java:294)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//me.saiintbrisson.minecraft.AbstractView.scheduleUpdate(AbstractView.java:475)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//me.saiintbrisson.minecraft.AbstractVirtualView.scheduleUpdate(AbstractVirtualView.java:300)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//fr.olten.xmas.clan.view.ClanView.<init>(ClanView.java:24)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//fr.olten.xmas.commands.ClanCommand.clan(ClanCommand.java:37)
[10:42:12 ERROR]: [Economy] [ACF]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[10:42:12 ERROR]: [Economy] [ACF]       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[10:42:12 ERROR]: [Economy] [ACF]       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[10:42:12 ERROR]: [Economy] [ACF]       at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//co.aikar.commands.RegisteredCommand.invoke(RegisteredCommand.java:152)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//co.aikar.commands.BaseCommand.executeCommand(BaseCommand.java:577)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//co.aikar.commands.BaseCommand.execute(BaseCommand.java:513)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//co.aikar.commands.RootCommand.execute(RootCommand.java:99)
[10:42:12 ERROR]: [Economy] [ACF]       at xmas-economy-0.1.0.jar//co.aikar.commands.BukkitRootCommand.execute(BukkitRootCommand.java:84)
[10:42:12 ERROR]: [Economy] [ACF]       at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
[10:42:12 ERROR]: [Economy] [ACF]       at org.bukkit.craftbukkit.v1_19_R1.CraftServer.dispatchCommand(CraftServer.java:907)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.network.PlayerConnection.handleCommand(PlayerConnection.java:2376)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$18(PlayerConnection.java:2147)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.TickTask.run(TickTask.java:18)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1358)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:183)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.MinecraftServer.bh(MinecraftServer.java:1335)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1328)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1306)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.MinecraftServer.v(MinecraftServer.java:1191)
[10:42:12 ERROR]: [Economy] [ACF]       at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:302)
[10:42:12 ERROR]: [Economy] [ACF]       at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by DevNatan** [comment]
@Azodox ya, it's a bug, will be hotfixed on v2.5.2.

As a recommendation use `slot(...).onRender` instead of `slot(...).onUpdate`, the rendering function has priority over the update and it is also called during the update, it is only used o the update function if you want a specific special handling for actions during the update, it should always be used in conjunction with onRender or in conjunction with a fallback item.

Also, render is called during the first render, not the update which means your items will only appear in the container for the player a second later (with the scheduleUpdate).

Ex.: we have a fallback item and when it updates, switch to anothr item
```java
slot(fallbackItem).onUpdate(update -> update.setItem(someCondition ? updateItem : null));
```

**Comment by DevNatan** [comment]
@Azodox Fixed in 9c8ed090c9.

You can use the commit as version of IF dependency if u shade it in your project or use the [generated artifacts](https://github.com/DevNatan/inventory-framework/actions/runs/2694023421) (scroll down) for your platform if not.
